newbank/server/Loan.java

```diff
@@ -47,13 +51,27 @@ public double getInitialAmount() {
     }
 
     public double getAmount() {
+        return amount;
+    }
+
+    public double getAmountWithInterest(){
         LocalDate now = currentTime.get();
         for (LocalDate i = this.timeOfLastInterestPayment; i.isBefore(now); i = i.plusDays(1)) {
-          double dailyInterest = getDailyInterest();
-          this.amount += dailyInterest;
+            double dailyInterest = getDailyInterest();
+            this.interestAmount += dailyInterest;
+
+            if(interestDueDate == i){
+                this.interestDueDate = interestDueDate.plusDays(365);
+                this.amount += this.interestAmount;
+                this.interestAmount = 0;
+            }
         }
         this.timeOfLastInterestPayment = now;
-        return amount;
+        return this.amount + this.interestAmount;
+    }
+
+    public double getEndAmount(){
```

Comment:
Good work. One hint which could be improved: If the interest exceeds one year, compound interest should be included. Could you optimise this? ;-).

Reply:
That's the way to go, thank you for improving the routine.